casper/src/main/scala/coop/rchain/casper/LastFinalizedBlockCalculator.scala

```diff
@@ -8,14 +8,12 @@ import coop.rchain.casper.util.ProtoUtil
 import coop.rchain.shared.{Cell, Log}
 import coop.rchain.catscontrib.ListContrib
 
-object LastFinalizedBlockCalculator {
-  // TODO: Extract hardcoded fault tolerance threshold
-  private val faultToleranceThreshold = 0f
-
-  def run[F[_]: Monad: Log: Sync: Concurrent: BlockStore: BlockDagStorage: SafetyOracle](
-      dag: BlockDagRepresentation[F],
-      lastFinalizedBlockHash: BlockHash
-  )(implicit state: Cell[F, CasperState]): F[BlockHash] =
+final case class LastFinalizedBlockCalculator[F[_]: Sync: Log: Concurrent: BlockStore: BlockDagStorage: SafetyOracle](
+    faultToleranceThreshold: Float
+) {
+  def run(dag: BlockDagRepresentation[F], lastFinalizedBlockHash: BlockHash)(
+      implicit state: Cell[F, CasperState]
```

Comment:
Maybe it would be a good idea to introduce a type alias for `Cell[F, CasperState]`?